### PR TITLE
added on hold status

### PR DESCRIPTION
**Pull request by ravikantkumar6**

### CODE REVIEW

File: common/src/main/java/com/publicissapient/kpidashboard/common/model/application/FieldMapping.java [hunk 1]
```diff
@@ -175,6 +175,10 @@ public class FieldMapping extends BasicModel {
 
 	// issue status to exclude missing worklogs
 	private List<String> issueStatusExcluMissingWork;
+
+	// issue On Hold status to exclude Closure possible
+	private List<String> issueOnHoldStatusExcluClosurePossible;
+
```
Comment (hirbabar) [inline common/src/main/java/com/publicissapient/kpidashboard/common/model/application/FieldMapping.java:181]: added new field then have to added in this file also kpi_fieldmapping_insert.js

**Review by hirbabar — APPROVED** [review]

File: common/src/main/java/com/publicissapient/kpidashboard/common/model/application/FieldMapping.java (hunk 1 above)
Comment (ravikantkumar6) [reply common/src/main/java/com/publicissapient/kpidashboard/common/model/application/FieldMapping.java:181]: added

File: mongodb/scripts/kpi_fieldmapping_insert.js [hunk 2]
```diff
@@ -363,7 +363,7 @@ db.getCollection('kpi_fieldmapping').insert(
         kpiName: 'Closure Possible Today',
 		kpiSource: 'Jira',
         type: ['Other'],
-        fieldNames : { 'Workflow Status Mapping' : ['jiraStatusForQa'], 'Custom Fields Mapping' : ['workingHoursDayCPT'] }
+        fieldNames : { 'Workflow Status Mapping' : ['jiraStatusForQa'], 'Custom Fields Mapping' : ['workingHoursDayCPT'] , 'Workflow Status Mapping' : ['jiraStatusForQa']}
       },
```
Comment (hirbabar) [inline mongodb/scripts/kpi_fieldmapping_insert.js:367]: wrong field name added

**Review by hirbabar — APPROVED** [review]

File: UI/src/app/config/project-config/field-mapping/field-mapping.component.html [hunk 3]
```diff
@@ -298,6 +298,21 @@ <h5 class="card__primary__title__text p-m-0">Field Mappings (Scrum)</h5>
...
+                    <div class="p-mb-3 p-d-flex p-align-center"
+                        *ngIf="fieldstoShow?.includes('issueOnHoldStatusExcluClosurePossible') || fieldstoShow.length === 0">
+                        <label class="field-title">Issue On hold status to be excluded from Closure possible <span class="fa fa-info-circle" aria-hidden="true" pTooltip='<span>
+                            Issue tagged to the workflow status mentioned in this field should be excluded from Closure possible today KPI
+                                               </span>' tooltipPosition="top" tooltipEvent="hover"
+                                tooltipStyleClass="tooltiptext" [escape]="false"></span></label>
+                        <div class="p-inputgroup">
+                            <p-chips [style]="{'max-width':'initial','width':'60vw', 'border-bottom-right-radius':'0'}"
+                                inputId="issueOnHoldStatusExcluClosurePossible" formControlName="issueOnHoldStatusExcluClosurePossible"
+                                placeholder="Enter Status to Identify 'In Testing' status" addOnBlur="true">
```
Comment (hirbabar) [inline UI/src/app/config/project-config/field-mapping/field-mapping.component.html:310]: changes label/placeholder/inputId name as On hold status (jiraOnHoldStatus)

File: UI/src/app/config/project-config/field-mapping/field-mapping.component.ts [hunk 4]
```diff
@@ -405,6 +405,7 @@ export class FieldMappingComponent implements OnInit {
       jiraDor: [''],
       jiraDefectRejectionStatus: [''],
       issueStatusExcluMissingWork: [],
+      issueOnHoldStatusExcluClosurePossible: [],
```
Comment (hirbabar) [inline UI/src/app/config/project-config/field-mapping/field-mapping.component.ts:408]: jiraOnHoldStatus - make it generalize for another kpi uses

**Review by hirbabar — APPROVED** [review]

**Pull request merged**